gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GHFSInstrumentation.java

```diff
@@ -0,0 +1,331 @@
+package com.google.cloud.hadoop.fs.gcs;
+
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.statistics.*;
+import org.apache.hadoop.fs.statistics.impl.IOStatisticsBinding;
+import org.apache.hadoop.fs.statistics.impl.IOStatisticsStore;
+import org.apache.hadoop.fs.statistics.impl.IOStatisticsStoreBuilder;
+import org.apache.hadoop.metrics2.*;
+import org.apache.hadoop.metrics2.impl.MetricsSystemImpl;
+import org.apache.hadoop.metrics2.lib.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.net.URI;
+import java.time.Duration;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.apache.hadoop.fs.statistics.StoreStatisticNames.ACTION_HTTP_GET_REQUEST;
+import static org.apache.hadoop.fs.statistics.StoreStatisticNames.SUFFIX_FAILURES;
+
+public class GHFSInstrumentation implements Closeable, MetricsSource, IOStatisticsSource, DurationTrackerFactory{
```

Comment:
Add Java doc

Reply:
Make this class package-private if possible.

Reply:
We cannot make this package private as we are using this in GcsDelegationTokens to track delegation related statistics